extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -719,13 +721,162 @@ class SegmentationService {
+    segmentationId = segmentationId ?? rtDisplaySet.displaySetInstanceUID;
+    const { structureSet } = rtDisplaySet;
+
+    if (!structureSet) {
+      throw new Error(
+        'To create the contours from RT displaySet, the displaySet should be loaded first, you can perform rtDisplaySet.load() before calling this method.'
+      );
+    }
+
+    const defaultScheme = this._getDefaultSegmentationScheme();
+    const rtDisplaySetUID = rtDisplaySet.displaySetInstanceUID;
+
+    const allRTStructData = structureSet.ROIContours.map(
+      ({ contourPoints, ROINumber, ROIName, colorArray }) => {
+        const data = contourPoints.map(({ points, ...rest }) => {
+          const newPoints = points.map(({ x, y, z }) => {
+            return [x, y, z];
+          });
+
+          return {
+            ...rest,
+            points: newPoints,
+          };
+        });
+
+        return {
+          data,
+          id: ROIName || ROINumber,
+          segmentIndex: ROINumber,
+          color: colorArray,
+          geometryId: `${rtDisplaySetUID}:${ROIName ||
+            ROINumber}:segmentIndex-${ROINumber}`,
+        };
+      }
+    );
+
+    // sort by segmentIndex
+    allRTStructData.sort((a, b) => a.segmentIndex - b.segmentIndex);
+
+    const geometryIds = allRTStructData.map(({ geometryId }) => geometryId);
+
+    const segmentation: Segmentation = {
+      ...defaultScheme,
+      id: segmentationId,
+      displaySetInstanceUID: rtDisplaySetUID,
+      type: representationType,
+      representationData: {
+        [CONTOUR]: {
+          geometryIds,
+        },
+      },
+    };
+
+    const cachedSegmentation = this.getSegmentation(segmentationId);
+
+    if (cachedSegmentation) {
+      // if the labelmap with the same segmentationId already exists, we can
+      // just assume that the segmentation is already created and move on with
+      // updating the state
+      return this.addOrUpdateSegmentation(
+        Object.assign(segmentation, cachedSegmentation),
+        suppressEvents
+      );
+    }
+
+    if (!structureSet.ROIContours?.length) {
+      throw new Error(
+        'The structureSet does not contain any ROIContours. Please ensure the structureSet is loaded first.'
+      );
+    }
+    let count = 0;
+    const segmentsCachedStats = {};
+    const initializeContour = async rtStructData => {
+      const { data, id, color, segmentIndex, geometryId } = rtStructData;
+      const geometry = await geometryLoader.createAndCacheGeometry(geometryId, {
+        geometryData: {
+          data,
+          id,
+          color,
+          frameOfReferenceUID: structureSet.frameOfReferenceUID,
+          segmentIndex,
+        },
+        type: csEnums.GeometryType.CONTOUR,
+      });
+
+      const contourSet = geometry.data;
+      const centroid = contourSet.getCentroid();
+
+      segmentsCachedStats[segmentIndex] = {
+        center: { world: centroid },
+        modifiedTime: Date.now(),
```

Comment:
This is a bad idea to set the modified time to now - really, this is an unmodified RT that happens to have just been loaded.  Setting it to the current time means one can't distinguish updates from loads, and that prevents easy display of things like save required indicators.